### ISSUE TITLE
poudriere bulk, Pressing ctrl-c does not  unmount everything.

### ISSUE THREAD

**Issue by devosalain** [freebsd/poudriere#1042]
After launching poudriere bulk, pressing Ctrl-c does sometimes not unmount everything.
This happens in 40% off the cases. In 60% the unmount of everything works fine.
What is still mounted can be seen with "zfs mount" & "mount".
Manual unmount fails , so to be certain i do a reboot.

**Comment by bdrewery** [comment]
Is this in poudriere-devel or release?

**Comment by infostud** [comment]
I find I have to reboot when a run fails or is cancelled as mounts are left in jails. With the next run (I use buildbot for CI)

    sudo poudriere ports -u -p default -v
Fails with message

    [00:00:00] Error: Ports tree "default" is currently mounted and being used.
I'm using Poudriere 3.3.7_1.
Could there be a command to clean up after a cancelled or failed run?

**Comment by nunotexbsd** [comment]
> I find I have to reboot when a run fails or is cancelled as mounts are left in jails. With the next run (I use buildbot for CI)
> 
> ```
> sudo poudriere ports -u -p default -v
> ```
> 
> Fails with message
> 
> ```
> [00:00:00] Error: Ports tree "default" is currently mounted and being used.
> ```
> 
> I'm using Poudriere 3.3.7_1. Could there be a command to clean up after a cancelled or failed run?

`poudriere jail --k ${jail}`      Kill a jail (stop it).
Is that what you looking for?

**Comment by infostud** [comment]
Thank you for your suggestion but things haven't improved. I'm confused about the message below ${jail}-default not running

    root@vcobie:~ # poudriere jail -l
    131amd64 13.1-RELEASE amd64 http   2022-05-17 16:35:36 /usr/local/poudriere/jails/131amd64
    root@vcobie:~ # poudriere jail -k 131amd64
    Missing: JAILNAME
    ... (usage info)
    root@vcobie:~ # poudriere jail -k -j 131amd64
    [00:00:00] Jail 131amd64-default not running, but cleaning up anyway
    [00:00:00] Unmounting file systems
    root@vcobie:~ # poudriere ports -u -p default -v
    [00:00:00] Error: Ports tree "default" is currently mounted and being used.
    root@vcobie:~ # poudriere jail -l
    131amd64 13.1-RELEASE amd64 http   2022-05-17 16:35:36 /usr/local/poudriere/jails/131amd64
    root@vcobie:~ # jls
    JID  IP Address      Hostname                      Path
    83  127.0.0.1       131amd64-default-workstation  /usr/local/poudriere/data/.m/131amd64-default-workstation/ref
    84                  131amd64-default-workstation  /usr/local/poudriere/data/.m/131amd64-default-workstation/ref
    85  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/03
    86                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/03
    87  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/02
    88                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/02
    89  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/01
    90  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/04
    91                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/01
    92                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/04

**Comment by nunotexbsd** [comment]
> Thank you for your suggestion but things haven't improved. I'm confused about the message below ${jail}-default not running
> 
> ```
> root@vcobie:~ # poudriere jail -l
> 131amd64 13.1-RELEASE amd64 http   2022-05-17 16:35:36 /usr/local/poudriere/jails/131amd64
> root@vcobie:~ # poudriere jail -k 131amd64
> Missing: JAILNAME
> ... (usage info)
> root@vcobie:~ # poudriere jail -k -j 131amd64
> [00:00:00] Jail 131amd64-default not running, but cleaning up anyway
> [00:00:00] Unmounting file systems
> root@vcobie:~ # poudriere ports -u -p default -v
> [00:00:00] Error: Ports tree "default" is currently mounted and being used.
> root@vcobie:~ # poudriere jail -l
> 131amd64 13.1-RELEASE amd64 http   2022-05-17 16:35:36 /usr/local/poudriere/jails/131amd64
> root@vcobie:~ # jls
> JID  IP Address      Hostname                      Path
> 83  127.0.0.1       131amd64-default-workstation  /usr/local/poudriere/data/.m/131amd64-default-workstation/ref
> 84                  131amd64-default-workstation  /usr/local/poudriere/data/.m/131amd64-default-workstation/ref
> 85  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/03
> 86                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/03
> 87  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/02
> 88                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/02
> 89  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/01
> 90  127.0.0.1       131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/04
> 91                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/01
> 92                  131amd64-default-workstation- /usr/local/poudriere/data/.m/131amd64-default-workstation/04
> ```

Missing -j switch

-- `poudriere jail -k 131amd64`
++ `poudriere jail -k -j 131amd64`

**Comment by infostud** [comment]
Thank you. In the code you quoted you'll see I added the -j parameter on the next command line after showing just the pertinent part of response and usage output from the incorrect command. I showed both the incorrect and correct command lines in case someone finds this thread in the future. Perhaps that was confusing.

    Missing: JAILNAME
    ... (usage info)
    root@vcobie:~ # poudriere jail -k -j 131amd64
    [00:00:00] Jail 131amd64-default not running, but cleaning up anyway
    [00:00:00] Unmounting file systems

Why does Poudriere say the jail I'm attempting to kill isn't running when jls shows it and subjails are? They are tying up mount points.
